### PR TITLE
Fix dependencies order

### LINKED ISSUE #58: prettier-package-json sorts dependencies differently than npm
In case a dependency has an uppercase letter in the name, the order of the dependencies might be different:

```
% npm init --yes         
Wrote to /Users/mykhalov/test/package.json:

{
  "name": "test",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "description": ""
}


% npm i Select2      

added 1 package in 1s
% npm i select2

added 1 package, and removed 1 package in 922ms
% cat package.json 
{
  "name": "test",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "description": "",
  "dependencies": {
    "select2": "^4.1.0-rc.0",
    "Select2": "^3.5.7"
  }
}
% npx prettier-package-json
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "license": "ISC",
  "author": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": {
    "Select2": "^3.5.7",
    "select2": "^4.1.0-rc.0"
  },
  "keywords": []
}
```

This is inconvenient because every time you install something, you also have to run prettier-package-json.

### PR DESCRIPTION
**Pull request by pavloconuve**
When sorting *dependencies fields, use localeCompare-based function to
match the result to those of npm

Fixes #58

### CODE REVIEW

**Review by cameronhunter — APPROVED** [review]
Thanks for the contribution!

**Comment by mykhalov** [comment]
@cameronhunter thanks for the quick response!

To be able to merge this, I'm also going to need maintainer's approval to run the workflow (https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks).

**Comment by cameronhunter** [comment]
I've approved and run the pipeline

**Comment by mykhalov** [comment]
@cameronhunter still can't merge it (no write access), passing over to you. Hope this will be released soon!

**Pull request merged**
